oeps/oep-0023-theming.rst

```diff
@@ -0,0 +1,204 @@
+      applications.
+
+2. This theming system will surface an API defining which elements can
+      be modified and how. As outlined in OEP-16 [7], this API will be
+      based on the Bootstrap framework.
+
+3. As consumers of the new theming API, the core contributor team will
+      make every reasonable effort to keep breaking changes to an
+      absolute minimum. If breaking changes must occur, the core
+      contributor team will enumerate them with every named release of
+      the Open edX platform.
+
+4. The core contributor team will provide official documentation,
+      examples, and tools to help other core contributors and open
+      source community members develop themes.
+
+5. Further modification to UI elements or content will not be officially
+      supported within the scope of this theming system. A system to
+      enable content modification may eventually be developed and
+      supported, but its definition is outside the scope of this
+      proposal.
+
+Plan
+----
+
+1. API definition for the new theming system will be proposed, agreed
+      upon, and documented.
+
+2. The core contributor team will develop an example theme compliant
+      with the new system and use it to build support into the platform.
+      Existing work [8] on such a theme already exists in the
+      edx-bootstrap repository.
+
+3. Open edX core and community contributors will build scaffolding to
+      support the new theming system throughout the Open edX platform.
+      This scaffolding will be implemented in as unobtrusive a means as
+      possible with all attempts made not to interfere with existing
+      theming systems.
+
+4. Once scaffolding is complete, the core contributor team will announce
+      an alpha release of the new theming system with the closest named
+      release of the Open edX platform.
+
+5. Throughout the lifecycle of the aforementioned release, the core
+      contributor team will solicit feedback and contributions and
+      respond to suggestions from community members willing to try out
+      the new theming system.
+
+6. By the next sequential named release of the Open edX platform,
+      assuming the new theming system has delivered value to its alpha
+      stakeholders, the core contributor team will announce a beta
+      release of the new theming system and announce the official
+      deprecation of all legacy theming systems.
+
+7. The Open edX core contributor team will continue to solicit
+      contributions and feedback on the theming system until the
+      following named release of the platform, by which point they will
+      announce official support for the new system.
+
+8. Removal of legacy theming-related code may begin two named releases
+      after the announcement of its deprecation.
+
+Consequences
+============
+
+-  All existing theming systems will be deprecated and support will
+      eventually be removed from the platform.
```

Comment:
I'm in favour of removing obsolete systems, but we need to have way to do content modification first.

Reply:
Definitely agree! My wording around some of this was unclear -- I'll leave a comment below, and clean up the language.